app/client/components/pageSkeleton.tsx

```diff
@@ -0,0 +1,177 @@
+  title: "Manage payment method",
+  path: `/payment/${productType.urlPart}`,
+  selectedNavItem: NAV_LINKS.accountOverview
+}));
+
+const holidaysOverviewLocationObjects: LocationObject[] = Object.values(
+  PRODUCT_TYPES
+)
+  .filter(shouldHaveHolidayStopsFlow)
+  .map(productType => ({
+    title: "Manage suspensions",
+    path: `/suspend/${productType.urlPart}`,
+    selectedNavItem: NAV_LINKS.accountOverview
+  }));
+
+const deliveryAddressFormLocationObjects: LocationObject[] = Object.values(
+  PRODUCT_TYPES
+)
+  .filter(shouldHaveHolidayStopsFlow)
+  .map(productType => ({
+    title: "Update delivery details",
+    path: `/delivery/${productType.urlPart}/address`,
+    selectedNavItem: NAV_LINKS.accountOverview
+  }));
+
+const deliveryRecordsLocationObjects: LocationObject[] = Object.values(
+  PRODUCT_TYPES
+)
+  .filter(shouldHaveHolidayStopsFlow)
+  .map(productType => ({
+    title: "Delivery history",
+    path: `/delivery/${productType.urlPart}/records`,
+    selectedNavItem: NAV_LINKS.accountOverview
+  }));
+
+const MMALocationObjectArr: LocationObject[] = [
+  {
+    title: "Account overview",
+    path: "/",
+    selectedNavItem: NAV_LINKS.accountOverview
+  },
+  {
+    title: "Billing",
+    path: "/billing",
+    selectedNavItem: NAV_LINKS.billing
+  },
+  ...manageProductLocationObjects,
+  ...cancellationFlowLocationObjects,
+  ...paymentUpdateFlowLocationObjects,
+  ...holidaysOverviewLocationObjects,
+  ...deliveryAddressFormLocationObjects,
+  ...deliveryRecordsLocationObjects,
+  {
+    title: "Emails & marketing",
+    path: "/email-prefs",
+    selectedNavItem: NAV_LINKS.emailPrefs
+  },
+  {
+    title: "Edit your profile",
+    path: "/public-settings",
+    selectedNavItem: NAV_LINKS.profile
+  },
+  {
+    title: "Settings",
+    path: "/account-settings",
+    selectedNavItem: NAV_LINKS.settings
+  },
+  {
+    title: "Help",
+    path: "/help",
+    selectedNavItem: NAV_LINKS.help
+  }
+];
+
+const nonMMALocationObjectArr: LocationObject[] = [
+  {
+    title: "Help Centre",
+    path: "/help-centre",
+    selectedNavItem: NAV_LINKS.help
+  },
+  {
+    title: "Need to contact us about something?",
+    path: "/contact-us",
+    selectedNavItem: NAV_LINKS.help
+  }
+];
+
+const PageSkeleton = () => {
+  return (
+    <Location>
+      {({ location }) => {
+        const selectedMMALocationObject = MMALocationObjectArr.filter(
+          currentObject => location.pathname === currentObject.path
+        )[0];
+        const selectedNonMMALocationObject = nonMMALocationObjectArr.filter(
+          currentObject => location.pathname.startsWith(currentObject.path)
+        )[0];
+
+        if (selectedNonMMALocationObject) {
+          return (
+            <>
+              <SectionHeader title={selectedNonMMALocationObject.title} />
+              <SectionPageContainer sectionTitle="&nbsp;">
+                <div
+                  css={css`
+                    margin-bottom: ${space[24]}px;
```

Comment:
How safe is this access by index?

Reply:
I believe there will be some refactoring done here once there is a bit more standardisation across the codebase. This works for now with what we have, on safety I will defer to @rBangay and @ripecosta.

Reply:
Very safe. It is made available by the Guardian Design System (Source components). Documentation for `space` [here](https://www.theguardian.design/2a1e5182b/p/449bd5-space)